x-pack/test/fleet_api_integration/apis/epm/get.ts

```diff
@@ -109,7 +109,21 @@ export default function (providerContext: FtrProviderContext) {
       expect(packageInfo.download).to.equal(undefined);
       await uninstallPackage(testPkgName, '9999.0.0');
     });
-
+    describe('Installed Packages', () => {
+      before(async () => {
+        await installPackage(testPkgName, testPkgVersion);
+      });
+      after(async () => {
+        await uninstallPackage(testPkgName, testPkgVersion);
+      });
+      it('Allows the fetching of installed packages', async () => {
+        const res = await supertest
+          .get(`/api/fleet/epm/packages/installed?nameQuery=apache`)
```

Comment:
It would be great to add more integration tests that use the different query params, to test that the generated ES query works as expected.

Reply:
Agreed. I did try to expand these initially but was hitting issues due to the state from other tests leaking in and making the assertions unreliable. I *think* I've fixed that now though (will explain more in an overall comment below).